cli/azd/pkg/containerapps/container_app.go

```diff
@@ -118,105 +161,151 @@ func (cas *containerAppService) AddRevision(
 	containerApp.Properties.Template = revision.Properties.Template
 	containerApp, err = cas.syncSecrets(ctx, subscriptionId, resourceGroupName, appName, containerApp)
 	if err != nil {
-		return fmt.Errorf("syncing secrets: %w", err)
+		return nil, fmt.Errorf("syncing secrets: %w", err)
 	}
 
 	// Update the container app
 	err = cas.updateContainerApp(ctx, subscriptionId, resourceGroupName, appName, containerApp)
 	if err != nil {
-		return fmt.Errorf("updating container app revision: %w", err)
+		return nil, fmt.Errorf("updating container app revision: %w", err)
 	}
 
-	// If the container app is in multiple revision mode, update the traffic to point to the new revision
-	if *containerApp.Properties.Configuration.ActiveRevisionsMode == armappcontainers.ActiveRevisionsModeMultiple {
-		newRevisionName := fmt.Sprintf("%s--%s", appName, *revision.Properties.Template.RevisionSuffix)
-		err = cas.setTrafficWeights(ctx, subscriptionId, resourceGroupName, appName, containerApp, newRevisionName)
-		if err != nil {
-			return fmt.Errorf("setting traffic weights: %w", err)
-		}
+	newRevisionName := fmt.Sprintf("%s--%s", appName, *revision.Properties.Template.RevisionSuffix)
+	newRevisionResponse, err := revisionsClient.GetRevision(ctx, resourceGroupName, appName, newRevisionName, nil)
+	if err != nil {
+		return nil, fmt.Errorf("getting new revision '%s': %w", newRevisionName, err)
 	}
 
-	return nil
+	return &newRevisionResponse.Revision, nil
 }
 
-func (cas *containerAppService) syncSecrets(
+func (cas *containerAppService) ValidateRevision(
```

Comment:
`Validate` may suggest that this validates that the revision is in a recognizable state, but this seems to be closer to "polling until revision is provisioned and healthy". What about `PollRevisionUntilDone`? And we should document what `Done` means

Reply:
I'd like to keep this more generic so that if there are additional validation scenarios we need to perform they can happen here.  Will adds docs to explain what we are waiting for.